server/util/consistent_hash/consistent_hash_test.go

```diff
@@ -109,6 +115,27 @@ func TestEvenLoadDistribution(t *testing.T) {
 	assert.Less(t, skew, 0.1)
 }
 
+func TestAgainstReference(t *testing.T) {
```

Comment:
can we add a test that does something like:

func TestGoldenSet(t *testing.T) {
  ch := consistent_hash.NewConsistentHash(consistent_hash.SHA256, 10000)
  require.NoError(t, ch.Set("a", "b", "c", "d"))
  // N.B. These values should never change -- if you have to change them it means
  // the consistent hash is broken and returning results in a different order than before.
  assert.Equal(t, reference.GetAllReplicas(""), []string{"a", "b", "c", "d"}
  assert.Equal(t, reference.GetAllReplicas("1"), []string{"c", "d", "a"}
  assert.Equal(t, reference.GetAllReplicas("2"), []string{"d", "a", "c"}
  assert.Equal(t, reference.GetAllReplicas("3"), []string{"a", "c", "d"}
}